src/fieldml_util_routines.f90

```diff
@@ -207,8 +207,6 @@ SUBROUTINE FIELDML_IO_FINALISE( FIELDML_INFO, ERR, ERROR, * )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_CONNECTIVITY_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_LAYOUT_HANDLES, ERR, ERROR, *999 )
       DEALLOCATE(FIELDML_INFO)
```

Comment:
This is great that you tidied up the finalise with a pointer but I just checked and the corresponding initialise doesn't use a pointer or allocate the fieldml info variable? I'm surprised it runs? Maybe have a look in LIST_INITIALISE/LIST_FIANLISE to see how it is done? (NOTE: these routines haven't been converted to CaMeL case yet.). Note the use of the 998 label in the initialise routine to skip the finalise call to deallocate on error.